Sources/MapboxCoreNavigation/Feedback/NavigationEventsManager.swift

```diff
@@ -101,10 +96,57 @@ open class NavigationEventsManager {
         }
     }
 
-    /// :nodoc: the internal lower-level mobile events manager is an implementation detail which should not be manipulated directly
-    private var mobileEventsManager: MMEEventsManager!
+    private let eventsAPI: EventsAPI
+    private let telemetryService: TelemetryService
+
+    private let accessToken: String
+
+    @available(*, deprecated, renamed: "init(activeNavigationDataSource:passiveNavigationDataSource:accessToken:)")
+    public convenience init(activeNavigationDataSource: ActiveNavigationEventsManagerDataSource? = nil,
+                         passiveNavigationDataSource: PassiveNavigationEventsManagerDataSource? = nil,
+                         accessToken possibleToken: String? = nil,
+                         mobileEventsManager: MMEEventsManager = .shared()) {
+        mobileEventsManager.disableLocationMetrics()
+        self.init(activeNavigationDataSource: activeNavigationDataSource, passiveNavigationDataSource: passiveNavigationDataSource, accessToken: possibleToken)
+    }
+
+    public required init(activeNavigationDataSource: ActiveNavigationEventsManagerDataSource? = nil,
+                         passiveNavigationDataSource: PassiveNavigationEventsManagerDataSource? = nil,
+                         accessToken possibleToken: String? = nil) {
+        accessToken = possibleToken ?? NavigationEventsManager.obtainAccessToken()
+
+        let options = NavigationEventsManager.createEventsServerOptions(accessToken: accessToken)
+        self.eventsAPI = EventsService.getOrCreate(for: options)
```

Comment:
@tatiana-yan could you please elaborate on this?

Reply:
If you use different tokens you'll get different events services. Otherwise, it will not be possible to use different tokens for different SDKs within 1 application. 
If you call `getOrCreate` multiple times with the same options you'll get the same (singleton) service

`EventsService::getOrCreate`:
https://github.com/mapbox/mapbox-sdk-common/blob/066660e942186891b1c768a6655b8b83d0913206/components/events_service/src/mapbox/common/events_service_impl.cpp#L21

`HashOptionsServiceStorage`:
https://github.com/mapbox/mapbox-sdk-common/blob/066660e942186891b1c768a6655b8b83d0913206/components/core/include/mapbox/common/hash_options_service_storage.hpp#L31